Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java

```diff
@@ -152,11 +229,66 @@ private void handlePayPalResult(PaymentMethodNonce paymentMethodNonce) {
             super.onPaymentMethodNonceCreated(paymentMethodNonce);
 
             PayPalFragmentDirections.ActionPayPalFragmentToDisplayNonceFragment action =
-                PayPalFragmentDirections.actionPayPalFragmentToDisplayNonceFragment(paymentMethodNonce);
+                    PayPalFragmentDirections.actionPayPalFragmentToDisplayNonceFragment(paymentMethodNonce);
             action.setTransactionAmount(amount);
             action.setDeviceData(deviceData);
 
             NavHostFragment.findNavController(this).navigate(action);
         }
     }
+
+    @OptIn(markerClass = ExperimentalBetaApi.class)
+    private void launchPayPalEditFIVault(String editVaultId) {
+        PayPalVaultEditRequest request = new PayPalVaultEditRequest(
+                editVaultId
+        );
+
+        payPalClient.createEditAuthRequest(requireContext(), request, (result) -> {
+            if (result instanceof PayPalVaultEditResponse.Failure) {
+                PayPalVaultEditResponse.Failure.Failure failure = (PayPalVaultEditResponse.Failure) result;
+                String correlationId = failure.getRiskCorrelationId();
+                //TODO: PayPalVaultErrorHandlingEditRequest and Analytics
+            }
+
+            if (result instanceof PayPalVaultEditResponse.ReadyToLaunch) {
+                PayPalVaultEditResponse.ReadyToLaunch success = (PayPalVaultEditResponse.ReadyToLaunch) result;
+
+                //TODO: Analytics
+                PayPalPendingRequestEditFi pendingRequest = payPalLauncher.launch(requireActivity(), success);
+                if (pendingRequest instanceof PayPalPendingRequestEditFi.Started) {
+                    storePendingRequestForEditFi((PayPalPendingRequestEditFi.Started) pendingRequest);
+                } else if (pendingRequest instanceof PayPalPendingRequestEditFi.Failure) {
+                    handleError(((PayPalPendingRequestEditFi.Failure) pendingRequest).getError());
+                }
+            }
+        });
+    }
+
+    @OptIn(markerClass = ExperimentalBetaApi.class)
+    private void launchEditFiErrorHandlingRequest(String editVaultId, String riskCorrelationId) {
+        PayPalVaultErrorHandlingEditRequest request = new PayPalVaultErrorHandlingEditRequest(
+                editVaultId,
+                riskCorrelationId
+        );
+
+        payPalClient.createEditAuthRequest(requireContext(), request, (result) -> {
+            if (result instanceof PayPalVaultEditResponse.Failure) {
+                PayPalVaultEditResponse.Failure.Failure failure = (PayPalVaultEditResponse.Failure) result;
+                String correlationId = failure.getRiskCorrelationId();
+                //TODO: PayPalVaultErrorHandlingEditRequest and Analytics
```

Comment:
We will want to remove all of these analytics TODOs from the demo app, they will be sent from the SDK